addons/web/static/src/views/view_service.js

```diff
@@ -118,7 +120,17 @@ export const viewService = {
                             const { arch, toolbar, id, filters, custom_view_id } = views[viewType];
                             const viewDescription = { arch, id, custom_view_id };
                             if (toolbar) {
-                                viewDescription.actionMenus = toolbar;
+                                const actionMenus = toolbar;
+                                if (actionMenus.action) {
+                                    actionMenus.action = actionMenus.action.filter(
+                                        (action) =>
+                                            !evaluateBooleanExpr(
+                                                action.binding_invisible,
+                                                user.evalContext
+                                            )
+                                    );
+                                }
+                                viewDescription.actionMenus = actionMenus;
```

Comment:
you do that in the cache, are you sure the evalContext never changes, or is it a mistake?

Reply:
It's done by design. Almost all the information in the `user.evalContext` comes from the `session_info`. The only information that can change is the `companies.active_ids`. And changing the companies will reload the web client.